### ISSUE TITLE
confounds and ImageQC file not generated when -combine_all_dwis was used 

### ISSUE THREAD

**Issue by weiwei-wch** [PennBBL/qsiprep#138]
Hi @mattcieslak 
   sorry for interrupting again, there is a crash log when I use -combine_all_dwis, all the other outputs look fine but there are no **confounds.tsv and **ImageQC_dwis.csv files producing, those files can sucessfully produce for both shelds seperatly if I remove the command, follows are the crash log:
```
Node: qsiprep_wf.single_subject_TEST004_wf.dwi_preproc_acq_multishell_wf.confounds_wf.concat
Working directory: /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/confounds_wf/concat

Node inputs:

denoising_confounds = /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/pre_hmc_wf/merge_and_denoise_wf/hstack_confounds/confounds.csv
fd = /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/confounds_wf/fdisp/fd_power_2012.txt
motion = /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/confounds_wf/add_motion_headers/eddy_correctedspm_rp_motion.tsv
original_bvals = ['/work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/pre_hmc_wf/merge_and_denoise_wf/merge_dwis/sub-TEST004_acq-multishell_merged.bval']
original_bvecs = ['/work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/pre_hmc_wf/merge_and_denoise_wf/merge_dwis/sub-TEST004_acq-multishell_merged.bvec']
original_files = ['/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii']
sliceqc_file = /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/hmc_sdc_wf/eddy/eddy_corrected.eddy_outlier_n_sqr_stdev_map

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/multiproc.py", line 344, in _send_procs_to_workers
    self.procs[jobid].run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 516, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 635, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 741, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 397, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/qsiprep/interfaces/confounds.py", line 61, in _run_interface
    newpath=runtime.cwd,
  File "/usr/local/miniconda/lib/python3.7/site-packages/qsiprep/interfaces/confounds.py", line 171, in _gather_confounds
    raise Exception("Gradients or original files don't match. File a bug report!")
Exception: Gradients or original files don't match. File a bug report!
```
thanks for your help!

**Comment by mattcieslak** [comment]
Hi @weiwei-wch, could you tell me which version you're using?

**Comment by weiwei-wch** [comment]
I use 0.8.0,

**Comment by weiwei-wch** [comment]
@mattcieslak 
I finish preprcessing with 0.9.0beta1, this problem is still existed, this is the command I use:
`docker run --rm -it -v /home/weiwei/Desktop/new_data_process/new_sequence/BIDS:/data -v /home/weiwei/Desktop/new_data_process/new_sequence/output:/out -v /home/weiwei/Desktop/new_data_process/new_sequence:/work pennbbl/qsiprep:0.8.0 /data /out participant --output-resolution 1.2 --fs-license-file /work/license.txt --do-reconall --force-spatial-normalization  --prefer_dedicated_fmaps --hmc-model eddy --eddy-config /work/eddy_params.json --work-dir /work`

and this is the crash log：
```
Node: qsiprep_wf.single_subject_TEST004_wf.dwi_preproc_acq_multishell_wf.confounds_wf.concat
Working directory: /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/confounds_wf/concat

Node inputs:

denoising_confounds = /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/pre_hmc_wf/merge_and_denoise_wf/merge_dwis/dwi_denoise_acq_multishell_0_merged_confounds.csv
fd = /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/confounds_wf/fdisp/fd_power_2012.txt
motion = /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/confounds_wf/add_motion_headers/eddy_correctedspm_rp_motion.tsv
original_bvals = ['/work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/pre_hmc_wf/merge_and_denoise_wf/merge_dwis/dwi_denoise_acq_multishell_0_merged.bval']
original_bvecs = ['/work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/pre_hmc_wf/merge_and_denoise_wf/merge_dwis/dwi_denoise_acq_multishell_0_merged.bvec']
original_files = ['/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-01_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii', '/data/sub-TEST004/dwi/sub-TEST004_acq-multishell_run-02_dwi.nii']
sliceqc_file = /work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/hmc_sdc_wf/eddy/eddy_corrected.eddy_outlier_n_sqr_stdev_map

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/multiproc.py", line 344, in _send_procs_to_workers
    self.procs[jobid].run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 516, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 635, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 741, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 397, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/qsiprep/interfaces/confounds.py", line 61, in _run_interface
    newpath=runtime.cwd,
  File "/usr/local/miniconda/lib/python3.7/site-packages/qsiprep/interfaces/confounds.py", line 171, in _gather_confounds
    raise Exception("Gradients or original files don't match. File a bug report!")
Exception: Gradients or original files don't match. File a bug report!
```

**Comment by mattcieslak** [comment]
could you send me the '/work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/pre_hmc_wf/merge_and_denoise_wf/merge_dwis/dwi_denoise_acq_multishell_0_merged.bval' file and '/work/qsiprep_wf/single_subject_TEST004_wf/dwi_preproc_acq_multishell_wf/pre_hmc_wf/merge_and_denoise_wf/merge_dwis/dwi_denoise_acq_multishell_0_merged.bvec' files?

**Comment by weiwei-wch** [comment]
sure, you can download them form [](https://pan.baidu.com/s/1dbp-QIiM1CRhkZzdqn7sMg) with code n9gt

**Comment by mattcieslak** [comment]
This should be fixed in 0.13.0RC1. Please re-open if not
